### PR TITLE
New Hive serde format

### LINKED ISSUE #90: Parquet table format issue in CDH5.2
The input/output formats and serde of hive tables created in CDH4.6  using Maestro was:
```
 ROW FORMAT SERDE\
  'parquet.hive.serde.ParquetHiveSerDe'\
STORED AS INPUTFORMAT\
  'parquet.hive.DeprecatedParquetInputFormat'\
OUTPUTFORMAT\
  'parquet.hive.DeprecatedParquetOutputFormat'\
```

But after the cluster upgrade to CDH5.2 all the existing table format changed to : 
```
ROW FORMAT SERDE\
  'org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe'\
STORED AS INPUTFORMAT\
  'org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat'\
OUTPUTFORMAT\
  'org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat’\
```

So while running the maestro job to load data to some of the existing tables ,the  job is getting failed throwing the exception:

"Table already exist but schema doesn't match"

### PR DESCRIPTION
**Pull request by stephanh**
Upgraded the ParquetTableDescriptor to use the new Hive Parquet Serde classes.

Also added tests to verify the compatibility of the tables created by us and using Hive DDL.

This closes #90.

### CODE REVIEW

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala [hunk 1]
```diff
@@ -94,13 +95,14 @@ Hive operations:
     val x = for {
       _  <- Hive.createParquetTable[SimpleHive]("test", "test", List("part1" -> "string", "part2" -> "string"), None)
       t1 <- Hive.existsTable("test", "test")
-      t2 <- Hive.existsTableStrict[SimpleHive]("test", "test", List("part1" -> "string"), None)
-      t3 <- Hive.existsTableStrict[SimpleHive]("test", "test", List("part1" -> "string", "part2" -> "string"), Some(new Path("/other")))
-      t4 <- Hive.existsTableStrict[SimpleHive]("test", "test", List("part1" -> "string", "part2" -> "string"), None, TextFormat())
-      t5 <- Hive.existsTableStrict[SimpleHive2]("test", "test", List("part1" -> "string", "part2" -> "string"), None)
-    } yield (t1, t2, t3, t4, t5)
+      t2 <- Hive.existsTableStrict[SimpleHive]("test", "test", List("part1" -> "string", "part2" -> "string"), None)
```
Comment (vineethvarghese) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala]: Was this just something we missed or is something in CDH5 forcing us to do this?

Comment (stephanh) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala]: It is a test I thought we missed but it's actually covered as part of the create test.

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/PartitionHiveParquetScroogeSpec.scala [hunk 2]
```diff
@@ -108,6 +109,30 @@ Hive support for (thrift scrooge encoded) partitioned parquet tables
...
+      """
+
+    Hive.query(ddl).run(hiveConf)
+    executesOk(write)
+
+    val path = hiveWarehouse </> s"$database.db" </> table
+
+    facts(
+      path ==> exists,
+      path</> s"p_domain=rdbms_changes" </> "*" </> "*" </> "*.parquet" ==> records(
```
Comment (SamRoberts) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/PartitionHiveParquetScroogeSpec.scala]: space between `path` and `</>`

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala [hunk 3]
```diff
@@ -158,6 +159,26 @@ Hive operations:
...
+      STORED AS PARQUET
+      """
+
+    val x = for {
+      _ <- Hive.createDatabase(db)
+      _ <- Hive.query(ddl)
+      t <- Hive.existsTableStrict[SimpleHive](db, table, List("part" -> "string"), None, ParquetFormat)
+    } yield t
+
+    x must beValue(true)
```
Comment (SamRoberts) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala]: `x must beTrue` ?

**Comment by SamRoberts** [comment]
:+1:

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala (hunk 3 above)
Comment (stephanh) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala]: No. `x` is a `Hive[Boolean`. `beValue(a)` evaluates the Hive action and ensures that the value of the `Result` is a `Ok(a)`.

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala (hunk 1 above)
Comment (stephanh) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala]: To make it more explicit I changed the weak `existsTable` to a `existsTableStrict`.

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/PartitionHiveParquetScroogeSpec.scala (hunk 2 above)
Comment (stephanh) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/PartitionHiveParquetScroogeSpec.scala]: Fixed.

**Comment by vineethvarghese** [comment]
:+1:

File: hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala (hunk 3 above)
Comment (SamRoberts) [inline hive/src/test/scala/au/com/cba/omnia/ebenezer/scrooge/hive/HiveSpec.scala]: Aha, fair enough. `beValue` sounds like a generic specs2 matcher at face value, though. Anyway: all good.

**Pull request merged**